mmv1/products/vertexai/IndexEndpointDeployedIndex.yaml

```diff
@@ -0,0 +1,295 @@
+        name: 'machineSpec'
+        description: The minimum number of replicas this DeployedModel will be always deployed on.
+        immutable: true
+        required: true
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'machineType'
+            description: |
+                        The type of the machine.
+
+                        See the [list of machine types supported for prediction](https://cloud.google.com/vertex-ai/docs/predictions/configure-compute#machine-types)
+
+                        See the [list of machine types supported for custom training](https://cloud.google.com/vertex-ai/docs/training/configure-compute#machine-types).
+
+                        For [DeployedModel](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.endpoints#DeployedModel) this field is optional, and the default value is n1-standard-2. For [BatchPredictionJob](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.batchPredictionJobs#BatchPredictionJob) or as part of [WorkerPoolSpec](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/CustomJobSpec#WorkerPoolSpec) this field is required.
+            immutable: true
+      - !ruby/object:Api::Type::Integer
+        name: 'minReplicaCount'
+        description: The minimum number of machine replicas this DeployedModel will be always deployed on. This value must be greater than or equal to 1.
+        required: true
+      - !ruby/object:Api::Type::Integer
+        name: 'maxReplicaCount'
+        description: The maximum number of replicas this DeployedModel may be deployed on when the traffic against it increases. If maxReplicaCount is not set, the default value is minReplicaCount
+        default_from_api: true
+  - !ruby/object:Api::Type::Boolean
+    name: enableAccessLogging
+    description: If true, private endpoint's access logs are sent to Cloud Logging.
+    immutable: true
+    default_value: false
+  - !ruby/object:Api::Type::NestedObject
+    name: deployedIndexAuthConfig
+    description: If set, the authentication is enabled for the private endpoint.
+    immutable: true
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: authProvider
+        description: Defines the authentication provider that the DeployedIndex uses.
+        properties:
+          - !ruby/object:Api::Type::Array
+            name: audiences
+            description: The list of JWT audiences. that are allowed to access. A JWT containing any of these audiences will be accepted.
+            immutable: true
+            item_type: Api::Type::String
+          - !ruby/object:Api::Type::Array
+            name: allowedIssuers
+            immutable: true
+            description: |
+              A list of allowed JWT issuers. Each entry must be a valid Google service account, in the following format: service-account-name@project-id.iam.gserviceaccount.com
+            item_type: Api::Type::String
```

Comment:
This falls under the same reasoning for setting the other fields as `immutable` which is that `mutate` can only be applied on `deployment_resources`. The only way to attempt an update is to include changes to the deployment resources, otherwise we get this error:
```hcl
│ Error: Error updating IndexEndpointDeployedIndex "projects/hc-terraform-testing/locations/us-central1/indexEndpoints/5511433975622533120/deployedIndex/index_endpoint_deployed_index_dedicated_resources_test_10": googleapi: Error 400: There is no deployment resources change found for update.
```

And even when we do include the deployment_resource update along with another field (in this case `deployed_index_auth_config` it'll attempt to send a `POST` for an update:
```hcl
 ---[ REQUEST ]
 POST /v1/projects/hc-terraform-testing/locations/us-central1/indexEndpoints/5511433975622533120:mutateDeployedIndex?alt=json
 Host: us-central1-aiplatform.googleapis.com
 User-Agent: Terraform/1.8.0-alpha20240131 
 Content-Length: 527
 Content-Type: application/json
 Accept-Encoding: gzip

 {
  "dedicatedResources": {
   "machineSpec": {
    "machineType": "n1-standard-32"
   },
   "maxReplicaCount": 8,
   "minReplicaCount": 3
  },
  "deployedIndexAuthConfig": {
   "authProvider": {
    "allowedIssuers": [
     "mauricio.alvarezleon@hashicorp.com"
    ],
    "audiences": [
     "123456-my-app"
    ]
   }
  },
  "deploymentGroup": "default",
  "displayName": "dedicated_resources_test
  "enableAccessLogging": true,
  "id": 
  "index": "projects/871647908372/locations/
  "reservedIpRanges": [
   "10.0.0.0"
  ]
 }
```

But will not perform an actual update on the resource itself. We can see this when it performs a `GET` request upon the operation returning done:
```hcl
 ---[ REQUEST ]
 GET /v1/projects/hc-terraform-testing/locations/us-central1/indexEndpoints/5511433975622533120?alt=json
 Host: us-central1-aiplatform.googleapis.com
 User-Agent: Terraform/1.8.0-alpha20240131 
 Content-Type: application/json
 Accept-Encoding: gzip


 
 2024/08/07 15:38:54 [DEBUG] Google API 
 ---[ RESPONSE ]
 HTTP/2.0 200 OK
 Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; 
 Cache-Control: private
 Content-Type: application/json; charset=UTF-8
 Date: Wed, 07 Aug 2024 22:38:54 GMT
 Server: ESF
 Vary: Origin
 Vary: X-Origin
 Vary: Referer
 X-Content-Type-Options: nosniff
 X-Frame-Options: SAMEORIGIN
 X-Xss-Protection: 0

 {
   "name": "projects/871647908372/locations/
   "displayName": "sample-endpoint",
   "description": "A sample vertex endpoint",
   "deployedIndexes": [
     {
       "id": 
       "index": "projects/871647908372/
       "displayName": 
       "createTime": "2024-07-30T20:32:09.
       "indexSyncTime": "2024-08-07T22:33:27.
       "enableAccessLogging": true,
       "reservedIpRanges": [
 "10.0.0.0"
       ],
       "deploymentGroup": "default",
       "dedicatedResources": {
         "machineSpec": {
           "machineType": "n1-standard-32"
         },
         "minReplicaCount": 3,
         "maxReplicaCount": 8
       }
     }
   ],
   "etag": 
   "labels": {
     "label-one": "value-one"
   },
   "createTime": "2024-07-30T20:31:47.
   "updateTime": "2024-07-30T20:32:04.
   "publicEndpointDomainName": "707325528.
   "encryptionSpec": {}
 }
```

It's worth noting that an update IS taking place, it just only applies to `deployment_resource` and we can see this in both requests where both `max_replica_count` and `min_replica_count` are updated where as `deployedIndexAuthConfig` is not found in the `GET` request